src/Serilog/Parameters/PropertyValueConverter.cs

```diff
@@ -121,11 +125,22 @@ LogEventPropertyValue CreatePropertyValue(object value, Destructuring destructur
                 return new ScalarValue(null);
 
             if (destructuring == Destructuring.Stringify)
-                return new ScalarValue(value.ToString());
+            {
+                return Stringify(value);
+            }
 
             var valueType = value.GetType();
             var limiter = new DepthLimiter(depth, _maximumDestructuringDepth, this);
 
+            if (destructuring == Destructuring.Destructure)
```

Comment:
One last thought - an interesting implementation option would be to actually create an `IScalarConversionPolicy` for strings specifically, and (if the limit has been set), insert the policy at the head of the collection of policies. Making insertion of the policy conditional would mean the feature was truly pay-for-play, i.e. if no maximum depth was set, no policy would appear in the list and so runtime cost would be zero (no need for this `if` block, type comparison, etc.).

The "stringify" case might not fall out nicely from this option, but we could definitely reconsider having it apply there.

A small added bonus would be that using `int?` for the depth (instead of the `MaxValue` sentinel) would be possible without having to do a `.Value` runtime check.

What do you think?